### DISCUSSION TITLE
Support modal routing

### DISCUSSION THREAD

**Discussion by matheusfernd** [vikejs/vike#1855] [Discussion]
It would be cool to extend Vike routing system to include routing by modals, and include it in the "navigate (import { navigate } from 'vike/client/router')" options and passing +Modal.tsx in the routes

**Comment by brillout** [comment]
What you want is a bit vague, feel free to elaborate. The more you elaborate the better. Maybe we can create a feature request then.

**Reply by matheusfernd** [reply]
Is there any way to pass a state into the navigate function?

**Reply by matheusfernd** [reply]
I thought about doing something similar to nextjs: https://nextjs.org/docs/app/building-your-application/routing/intercepting-routes

**Reply by brillout** [reply]
Created a feature request for it: https://github.com/vikejs/vike/issues/1862.

> Is there any way to pass a state into the navigate function?

I ain't sure that would be sufficient for implementing modal routes.
